client/my-sites/post-type-filter/index.jsx

```diff
@@ -3,44 +3,49 @@
  */
 import React, { PropTypes } from 'react';
 import { connect } from 'react-redux';
-import { find, includes, reduce } from 'lodash';
+import { compact, find, includes, reduce } from 'lodash';
 
 /**
  * Internal dependencies
  */
+import { areAllSitesSingleUser } from 'state/selectors';
 import { getSelectedSiteId } from 'state/ui/selectors';
-import { getSiteSlug, isJetpackSite } from 'state/sites/selectors';
-import { getNormalizedPostCounts } from 'state/posts/counts/selectors';
+import { isJetpackSite, isSingleUserSite, getSiteSlug } from 'state/sites/selectors';
+import { getNormalizedMyPostCounts, getNormalizedPostCounts } from 'state/posts/counts/selectors';
 import { mapPostStatus } from 'lib/route/path';
 import UrlSearch from 'lib/mixins/url-search';
 import QueryPostCounts from 'components/data/query-post-counts';
 import SectionNav from 'components/section-nav';
 import NavTabs from 'components/section-nav/tabs';
 import NavItem from 'components/section-nav/item';
 import Search from 'components/search';
+import AuthorSegmented from './author-segmented';
 
 const PostTypeFilter = React.createClass( {
 	mixins: [ UrlSearch ],
 
 	propTypes: {
+		authorToggleHidden: PropTypes.bool,
 		siteId: PropTypes.number,
-		query: PropTypes.object,
+		query: PropTypes.shape( {
+			author: PropTypes.number, // User ID
+			status: PropTypes.string,
+			type: PropTypes.string.isRequired,
+		} ).isRequired,
 		jetpack: PropTypes.bool,
 		siteSlug: PropTypes.string,
 		counts: PropTypes.object
 	},
 
-	getDefaultProps() {
-		return {
-			query: {}
-		};
-	},
-
 	getNavItems() {
-		const { query, siteSlug, jetpack, counts } = this.props;
+		const { query, siteId, siteSlug, jetpack, counts } = this.props;
 
 		return reduce( counts, ( memo, count, status ) => {
-			if ( ! jetpack && ! count && ! includes( [ 'publish', 'draft' ], status ) ) {
+			// * Always add 'publish' and 'draft' tabs
+			// * Add all tabs in all-sites mode
+			// * Add all tabs in JP mode, for CPTs
+			// * In all other cases, add status tabs only if there's at least one post/CPT with that status
+			if ( siteId && ! ( jetpack && query.type !== 'post' ) && ! count && ! includes( [ 'publish', 'draft' ], status ) ) {
```

Comment:
Do you think we could make that one more readable? By placing it in a separate utility method for example?

Reply:
I think I'd rather keep the criteria as close as possible to the actual rendering here. I think they may be subject to change for even more consistency between different modes and post types. When it comes to legibility, I think I'd start by changing the method to return an actual element tree (using `map` instead of `reduce`). In another PR 😉